server/blob/blob-sharing-file/src/test/java/org/apache/james/blob/sharing/file/SimpleBlobSharingMechanismTest.java

```diff
@@ -0,0 +1,132 @@
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.sharing.file;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+
+import javax.mail.Message;
+import javax.mail.internet.InternetAddress;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.blob.api.HashBlobId;
+import org.apache.james.blob.api.ObjectStoreException;
+import org.apache.james.blob.memory.MemoryBlobStore;
+import org.apache.james.filesystem.api.FileSystem;
+import org.apache.james.server.core.configuration.Configuration;
+import org.apache.james.server.core.filesystem.FileSystemImpl;
+import org.apache.james.util.MimeMessageUtil;
+import org.apache.mailet.base.MailAddressFixture;
+import org.apache.mailet.base.test.FakeMailContext;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class SimpleBlobSharingMechanismTest {
+    private static final byte[] BLOB_CONTENT = "blob_content".getBytes(StandardCharsets.UTF_8);
+
+    private BlobStore blobStore;
+    private FakeMailContext mailetContext;
+    private FileSystem fileSystem;
+    private SimpleBlobSharingMechanism testee;
+
+    @BeforeEach
+    void setUp() {
+        mailetContext = FakeMailContext.builder().postmaster(MailAddressFixture.POSTMASTER_AT_JAMES).build();
+        blobStore = new MemoryBlobStore(new HashBlobId.Factory());
+
+        Configuration configuration = Configuration.builder()
+            .workingDirectory("../")
+            .configurationFromClasspath()
+            .build();
+        fileSystem = new FileSystemImpl(configuration.directories());
+        SimpleBlobSharingMechanism.Configuration blobSharingConfiguration = new SimpleBlobSharingMechanism.Configuration("file://var/blobSharing");
+
+        testee = new SimpleBlobSharingMechanism(mailetContext, blobStore, fileSystem, blobSharingConfiguration);
+    }
+
+    @Test
+    void sharingBlobShouldSendAMail() {
+        BlobId blobId = blobStore.save(BLOB_CONTENT).block();
+
+        String explanation = "The content of a deleted message vault had been shared with you.";
```

Comment:
- I though we can share a bunch of mails to a sharee?
- s/`of a deleted message vault`/`of a deleted message`

Reply:
> I though we can share a bunch of mails to a sharee?

Don't confuse the mail we send to notify him and the content being shared please!

Reply:
> of a deleted message

No, we won't send an email per deleted message :trollface: 

Think of this as the LinShare of the poor.

Would you share the messages one by one in LinShare ?